pycfmodel/model/resources/generic_resource.py

```diff
@@ -1,32 +1,27 @@
-import logging
-from typing import ClassVar, Optional
+from typing import Optional
 
-from pydantic import Extra, validator
+from pydantic import ConfigDict, field_validator
 
 from pycfmodel.model.generic import Generic
 from pycfmodel.model.resources.resource import Resource
-from pycfmodel.model.resources.types import ResourceModels
-
-logger = logging.getLogger(__file__)
-
-_EXISTING_RESOURCE_TYPES = {klass.TYPE_VALUE for klass in ResourceModels.__args__}
 
 
 class GenericResource(Resource):
     """This class is used for all resource types that don't have a dedicated class."""
 
-    ALLOW_EXISTING_TYPES: ClassVar[bool] = True
-    Type: str
+    Type: Optional[str] = None  # Optional to handle cases like "AWS::CloudFormation::Authentication"
```

Comment:
@ignaciobolonio Review this case

Reply:
I don't have any issue with this as long as it has retrocompatibility